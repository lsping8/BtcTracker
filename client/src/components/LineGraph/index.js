import React, { Component } from 'react';
import chartjs from 'react-chartjs';
import chartOptions from './chartOptions'

const LineChart = chartjs.Line;

class LineGraph extends Component {
  render() {
		const { chartData,chartLabel } = this.props
		const graphData = {
			labels: chartLabel,
			datasets: [
				{
					label: "My First dataset",
					fillColor: "rgba(220,220,220,0.2)",
					strokeColor: "rgba(220,220,220,1)",
					pointColor: "rgba(220,220,220,1)",
					pointStrokeColor: "#fff",
					pointHighlightFill: "#fff",
					pointHighlightStroke: "rgba(220,220,220,1)",
					data: chartData
				}
			]
		};
    return (
      <div>
        <LineChart data={graphData} options={chartOptions} width="600" height="250"/>
      </div>
    );
  }
}

export default LineGraph;
