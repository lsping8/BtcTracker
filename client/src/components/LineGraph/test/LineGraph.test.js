import React from 'react'
import ReactDOM from 'react-dom'
import { shallow,configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import LineGraph from '../index'

configure({ adapter: new Adapter() });

const lineGraph = shallow(<LineGraph />)

describe('LineGraph', () => {
  it('renders without crashing', () => {
    expect(lineGraph).toMatchSnapshot()
  })
})
