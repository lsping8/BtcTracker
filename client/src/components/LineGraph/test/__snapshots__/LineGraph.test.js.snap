// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineGraph renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LineGraph />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <LineChart
        data={
                Object {
                        "datasets": Array [
                          Object {
                            "data": undefined,
                            "fillColor": "rgba(220,220,220,0.2)",
                            "label": "My First dataset",
                            "pointColor": "rgba(220,220,220,1)",
                            "pointHighlightFill": "#fff",
                            "pointHighlightStroke": "rgba(220,220,220,1)",
                            "pointStrokeColor": "#fff",
                            "strokeColor": "rgba(220,220,220,1)",
                          },
                        ],
                        "labels": undefined,
                      }
        }
        height="200"
        options={
                Object {
                        "bezierCurve": true,
                        "bezierCurveTension": 0.4,
                        "datasetFill": true,
                        "datasetStroke": true,
                        "datasetStrokeWidth": 2,
                        "legendTemplate": "<ul class=\\"<%=name.toLowerCase()%>-legend\\"><% for (var i=0; i<datasets.length; i++){%><li><span style=\\"background-color:<%=datasets[i].strokeColor%>\\"><%if(datasets[i].label){%><%=datasets[i].label%><%}%></span></li><%}%></ul>",
                        "offsetGridLines": false,
                        "pointDot": true,
                        "pointDotRadius": 4,
                        "pointDotStrokeWidth": 1,
                        "pointHitDetectionRadius": 20,
                        "responsive": true,
                        "scaleGridLineColor": "rgba(0,0,0,.05)",
                        "scaleGridLineWidth": 1,
                        "scaleShowGridLines": true,
                        "scaleShowHorizontalLines": true,
                        "scaleShowVerticalLines": true,
                      }
        }
        width="600"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": Object {
          "datasets": Array [
            Object {
              "data": undefined,
              "fillColor": "rgba(220,220,220,0.2)",
              "label": "My First dataset",
              "pointColor": "rgba(220,220,220,1)",
              "pointHighlightFill": "#fff",
              "pointHighlightStroke": "rgba(220,220,220,1)",
              "pointStrokeColor": "#fff",
              "strokeColor": "rgba(220,220,220,1)",
            },
          ],
          "labels": undefined,
        },
        "height": "200",
        "options": Object {
          "bezierCurve": true,
          "bezierCurveTension": 0.4,
          "datasetFill": true,
          "datasetStroke": true,
          "datasetStrokeWidth": 2,
          "legendTemplate": "<ul class=\\"<%=name.toLowerCase()%>-legend\\"><% for (var i=0; i<datasets.length; i++){%><li><span style=\\"background-color:<%=datasets[i].strokeColor%>\\"><%if(datasets[i].label){%><%=datasets[i].label%><%}%></span></li><%}%></ul>",
          "offsetGridLines": false,
          "pointDot": true,
          "pointDotRadius": 4,
          "pointDotStrokeWidth": 1,
          "pointHitDetectionRadius": 20,
          "responsive": true,
          "scaleGridLineColor": "rgba(0,0,0,.05)",
          "scaleGridLineWidth": 1,
          "scaleShowGridLines": true,
          "scaleShowHorizontalLines": true,
          "scaleShowVerticalLines": true,
        },
        "width": "600",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <LineChart
          data={
                    Object {
                              "datasets": Array [
                                Object {
                                  "data": undefined,
                                  "fillColor": "rgba(220,220,220,0.2)",
                                  "label": "My First dataset",
                                  "pointColor": "rgba(220,220,220,1)",
                                  "pointHighlightFill": "#fff",
                                  "pointHighlightStroke": "rgba(220,220,220,1)",
                                  "pointStrokeColor": "#fff",
                                  "strokeColor": "rgba(220,220,220,1)",
                                },
                              ],
                              "labels": undefined,
                            }
          }
          height="200"
          options={
                    Object {
                              "bezierCurve": true,
                              "bezierCurveTension": 0.4,
                              "datasetFill": true,
                              "datasetStroke": true,
                              "datasetStrokeWidth": 2,
                              "legendTemplate": "<ul class=\\"<%=name.toLowerCase()%>-legend\\"><% for (var i=0; i<datasets.length; i++){%><li><span style=\\"background-color:<%=datasets[i].strokeColor%>\\"><%if(datasets[i].label){%><%=datasets[i].label%><%}%></span></li><%}%></ul>",
                              "offsetGridLines": false,
                              "pointDot": true,
                              "pointDotRadius": 4,
                              "pointDotStrokeWidth": 1,
                              "pointHitDetectionRadius": 20,
                              "responsive": true,
                              "scaleGridLineColor": "rgba(0,0,0,.05)",
                              "scaleGridLineWidth": 1,
                              "scaleShowGridLines": true,
                              "scaleShowHorizontalLines": true,
                              "scaleShowVerticalLines": true,
                            }
          }
          width="600"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "datasets": Array [
              Object {
                "data": undefined,
                "fillColor": "rgba(220,220,220,0.2)",
                "label": "My First dataset",
                "pointColor": "rgba(220,220,220,1)",
                "pointHighlightFill": "#fff",
                "pointHighlightStroke": "rgba(220,220,220,1)",
                "pointStrokeColor": "#fff",
                "strokeColor": "rgba(220,220,220,1)",
              },
            ],
            "labels": undefined,
          },
          "height": "200",
          "options": Object {
            "bezierCurve": true,
            "bezierCurveTension": 0.4,
            "datasetFill": true,
            "datasetStroke": true,
            "datasetStrokeWidth": 2,
            "legendTemplate": "<ul class=\\"<%=name.toLowerCase()%>-legend\\"><% for (var i=0; i<datasets.length; i++){%><li><span style=\\"background-color:<%=datasets[i].strokeColor%>\\"><%if(datasets[i].label){%><%=datasets[i].label%><%}%></span></li><%}%></ul>",
            "offsetGridLines": false,
            "pointDot": true,
            "pointDotRadius": 4,
            "pointDotStrokeWidth": 1,
            "pointHitDetectionRadius": 20,
            "responsive": true,
            "scaleGridLineColor": "rgba(0,0,0,.05)",
            "scaleGridLineWidth": 1,
            "scaleShowGridLines": true,
            "scaleShowHorizontalLines": true,
            "scaleShowVerticalLines": true,
          },
          "width": "600",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
